docs/visualizing-your-sql-queries/plotting-with-ggplot.md

```diff
@@ -1,14 +1,638 @@
+(
+    ggplot(
+        table="ggplot_hist_q3",
+        with_="ggplot_hist_q3",
+        mapping=aes(
+            x=["status", "region"],
+            fill=["#008080", "#d500f9"],
+            color="black",
+        ),
+    )
+    + geom_histogram(bins=10)
+)
+```
+
+</details>
+<!-- #endregion -->
+
+<!-- #region -->
+
+### `facet_wrap`
+
+Histograms produced with `ggplot` API can also be arranged in a sequence of panels into a 2D grid, which is beneficial when dealing with a single variable that has multiple levels, or when you want to arrange the plots in a more space efficient manner.
+
+For example, the histogram for `payments` filled by all eight regions is tough to make sense of in a single plot. Therefore, `facet_wrap` can make visualizing the individual histograms easy:
+
+```{code-cell} ipython3
+plt.rcParams["figure.figsize"] = (15, 6)  # increase size of canvas
+
+(
+    ggplot(
+        table="ggplot_CTE",
+        with_="ggplot_CTE",
+        mapping=aes(
+            x="payments",
+        ),
+    )
+    + geom_histogram(bins=10)
+    + facet_wrap("region", legend=False)
+)
+```
+
+### Question 4 (Medium)
+
+Produce the same faceted histograms as above, but account for `fill` mapping to `status`. Make sure the legend is presented in the plots.
+
+<b>Hint:</b> To present the legends clearly and to not distort the histograms because of the legends, specify `plt.rcParams["figure.figsize"] = (15,7)` before the `ggplot` function.
+
+<!-- #region -->
+<details>
+<summary>Answers</summary>
+
+```{code-cell} ipython3
+plt.rcParams["figure.figsize"] = (15, 7)  # increase size of canvas
+
+(
+    ggplot(
+        table="ggplot_CTE",
+        with_="ggplot_CTE",
+        mapping=aes(x="payments"),
+    )
+    + geom_histogram(bins=10, fill="status")
+    + facet_wrap("region")
+)
+```
+
+</details>
+<!-- #endregion -->
+
+<!-- #region -->
+
+## Interactive `ggplot`
+
+Similar to the Interactive Queries and Parameterization [module](https://ploomber-sql.readthedocs.io/en/latest/interactive-queries-and-parameterization/introduction-to-ipywidgets.html#putting-it-together), we can use the use the [interact](https://ipywidgets.readthedocs.io/en/stable/examples/Using%20Interact.html#using-interact) API from [Jupyter Widgets](https://ipywidgets.readthedocs.io/en/stable/index.html#jupyter-widgets). Widgets can be used with either `geom_boxplot` or `geom_histogram`, with the latter providing scope for greater flexibility due to the `fill` and `color` mappings.
+
+Interact autogenerates UI controls for function arguments, and then calls the function with those arguments when you manipulate the controls interactively.
+
+To use interact, you need to define:
+
+1. Widgets to be controlled
+
+2. The plot function that includes `ggplot` with dynamic argument(s) specified with the respective widget variable
+
+3. Invoke `interact()` API
+
+Let’s see examples below!
+
+#### Histogram - Basic Usage (with Dropdown and Slider widgets)
```

Comment:
I get this warning `plotting-with-ggplot.md:445: WARNING: Non-consecutive header level increase; H2 to H4 [myst.header]` when building with `jupyter-book build docs`. Please change this header to H3 with "### Histogram - Basic Usage...."

Reply:
resolved, good to close this comment